framework/Supports/Autoptimize.php

```diff
@@ -130,6 +129,50 @@ public function add_dns_prefetch( $content ) {
 		return $content;
 	}
 
+	/**
+	 * Add target and rel for links.
+	 *
+	 * @param string $content HTML content generated for the page.
+	 *
+	 * @return string
+	 */
+	public function add_target_rel( $content ) {
+		$tags_matches = [];
+
+		if ( preg_match_all( '#<a.*>.*<\/a>#Usmi', $content, $matches ) ) {
+			$tags_matches = array_merge( $tags_matches, $matches[0] );
+		}
+
+		if ( empty( $tags_matches ) ) {
+			return $content;
+		}
+
+		preg_match( '#http(s)?\:\/\/(([a-z0-9\_\-\.]+)\.([a-z0-9]{2,5}))\/?#', site_url(), $site_domain );
```

Comment:
why you use a regular expression to get the domain?
there is a native function parse_url(). using regular expression here is not the best way.